src/lib/flight_tasks/tasks/Orbit/FlightTaskOrbit.cpp

```diff
@@ -244,25 +245,17 @@ void FlightTaskOrbit::generate_circle_yaw_setpoints(Vector2f center_to_position)
 		break;
 
 	case orbit_status_s::ORBIT_YAW_BEHAVIOUR_HOLD_FRONT_TANGENT_TO_CIRCLE:
-		if (_r > 0) {
-			_yaw_setpoint = atan2f(center_to_position(1), center_to_position(0)) + M_PI_F / 2.f;
-
-		} else {
-			_yaw_setpoint = atan2f(center_to_position(1), center_to_position(0)) - M_PI_F / 2.f;
-		}
-
+		_yaw_setpoint = wrap_pi(atan2f(center_to_position(1), center_to_position(0)) + (sign(_v) * M_PI_F / 2.f));
```

Comment:
Interesting that it survived without the `wrap_pi` before

Reply:
It did because the setpoint goes into sin/cos and hence gets implicitly wrapped with pi during further processing but in the log it showed according to the message specs out of range values.